### ISSUE TITLE
Unrecoverable syntax error

### ISSUE THREAD

**Issue by alexandre1985** [SublimeLinter/SublimeLinter-eslint#215]
I have got this code:
```
import React from 'react';
import './App.css';

const App = () => {
  return (
    <div className="App">Hello, world!</div>
  );
};

export default App;

```

And on the </div> I get this error message:

> W: 7 E: 2, Unclosed regular expression.; Unrecoverable syntax error (54% scanned)., Line 6, Column: 40.

I believe this is a problem from your linter since executing the `eslint` command doesn't give this error

**Comment by braver** [comment]
[Debug console output](https://github.com/SublimeLinter/SublimeLinter/blob/master/docs/troubleshooting.rst#debug-mode) or it didn't happen!

**Comment by alexandre1985** [comment]
Here you go:
```
DPI scale: 1
startup, version: 3143 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/daniel/.config/sublime-text-3/Packages
state path: /home/daniel/.config/sublime-text-3/Local
zip path: /opt/sublime_text/Packages
zip path: /home/daniel/.config/sublime-text-3/Installed Packages
ignored_packages: ["Vintage"]
Errors parsing theme:
icon_file_type is missing layer0.opacity, setting to 1.0 for backwards compatibility
pre session restore time: 0.335219
font face "Monospace" has different widths for bold characters, disabling to prevent text reflow
startup time: 0.449367
first paint time: 0.461953
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin A File Icon.A File Icon
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin ApplySyntax.ApplySyntax
reloading plugin ApplySyntax.support
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin Colorsublime.colorsublime-plugin
reloading plugin Emmet.emmet-plugin
reloading plugin Git.git_commands
reloading plugin Material Theme.Icons
reloading plugin Material Theme.MT
reloading plugin PHP CS Fixer.SublimePhpCsFixer
reloading plugin PHP Companion.PHP Companion
reloading plugin PHP Getters and Setters.php-getter-setter
==== USER TEMPLATES ===
reloading plugin PHP Getters and Setters.user_templates
==== USER TEMPLATES ===
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SnippetMaker.SnippetMaker
reloading plugin SublimeLinter-contrib-sass-lint.linter
reloading plugin SublimeLinter-cppcheck.linter
reloading plugin SublimeLinter-eslint.linter
reloading plugin SublimeLinter-jshint.linter
reloading plugin SublimeLinter-php.linter
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.log_handler
reloading plugin SublimeLinter.message_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin SublimeLinter.tooltips_view
reloading plugin rsub.rsub
reloading plugin Babel.Babel
reloading plugin SublimeCodeIntel.SublimeCodeIntel
reloading plugin SublimeCodeIntel.ordereddict
plugins loaded
[rsub] Server running on localhost:58698...
[PHP Getters and Setters] Registered template : 'PSR2'
[PHP Getters and Setters] Registered template : 'camelCase'
[PHP Getters and Setters] Registered template : 'camelCaseFluent'
[PHP Getters and Setters] Registered template : 'snakeCase'
[PHP Getters and Setters] Registered template : 'snakeCaseFluent'
[PHP Getters and Setters] ignored type hinting var types ['mixed', 'int', 'integer', 'double', 'float', 'number', 'string', 'boolean', 'bool', 'numeric', 'unknown']
[PHP Getters and Setters] template is  'PSR2'
[PHP Getters and Setters] register extra user templates []
[PHP Getters and Setters] ignoring visibility to getters and setters
[PHP Getters and Setters] setterBeforeGetter is False
SublimeLinter: SublimeLinter.sublime-package:71: Logging installed; log level INFO
SublimeLinter: SublimeLinter.sublime-package:62: debug mode: on
SublimeLinter: SublimeLinter.sublime-package:63: version: 4.0.5
Emmet: No need to update PyV8
SublimeLinter: SublimeLinter.sublime-package:103: PATH:
    /home/daniel/n/bin
    /usr/local/sbin
    /usr/local/bin
    /usr/bin
    /usr/lib/jvm/default/bin
    /usr/bin/site_perl
    /usr/bin/vendor_perl
    /usr/bin/core_perl
SublimeLinter: SublimeLinter.sublime-package:1115: jshint version query: /home/daniel/n/bin/jshint --version
SublimeLinter: SublimeLinter.sublime-package:1122: jshint version: 2.9.5
SublimeLinter: SublimeLinter.sublime-package:1089: jshint: (>= 2.5.0) satisfied by 2.9.5
SublimeLinter: SublimeLinter.sublime-package:256: detected syntax: javascript
SublimeLinter: #1 SublimeLinter.sublime-package:782: 'jshint' is linting 'App.js'
SublimeLinter: #1 SublimeLinter.sublime-package:1184: jshint: App.js ['/home/daniel/n/bin/jshint', '--verbose', '--filename', '/home/daniel/dev/react/hello-world/src/App.js', '-']
SublimeLinter: #2 SublimeLinter.sublime-package:782: 'eslint' is linting 'App.js'
SublimeLinter: #1 SublimeLinter.sublime-package:1187: jshint: cwd: /home/daniel/dev/react/hello-world
SublimeLinter: #2 SublimeLinter.sublime-package:1184: eslint: App.js ['/home/daniel/dev/react/hello-world/node_modules/.bin/eslint', '--format', 'json', '--stdin', '--stdin-filename', '/home/daniel/dev/react/hello-world/src/App.js']
SublimeLinter: #2 SublimeLinter.sublime-package:1187: eslint: cwd: /home/daniel/dev/react/hello-world
SublimeLinter: #1 SublimeLinter.sublime-package:809: jshint output:
    /home/daniel/dev/react/hello-world/src/App.js: line 6, col 40, Unclosed regular expression. (E015)
    /home/daniel/dev/react/hello-world/src/App.js: line 6, col 40, Unrecoverable syntax error. (54% scanned). (E041)

    2 errors
SublimeLinter: #1 SublimeLinter.sublime-package:850: jshint: No match for line: ''
SublimeLinter: #1 SublimeLinter.sublime-package:850: jshint: No match for line: '2 errors'
Could not import subprocess32 module, falling back to subprocess module
SublimeLinter: #2 SublimeLinter.sublime-package:809: eslint output:
    [{"filePath":"/home/daniel/dev/react/hello-world/src/App.js","messages":[{"ruleId":"react/jsx-quotes","severity":1,"message":"Definition for rule 'react/jsx-quotes' was not found","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"react/require-extension","severity":1,"message":"Definition for rule 'react/require-extension' was not found","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"react/wrap-multilines","severity":1,"message":"Definition for rule 'react/wrap-multilines' was not found","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"no-empty-label","severity":1,"message":"Rule 'no-empty-label' was removed and replaced by: no-labels","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"space-after-keywords","severity":1,"message":"Rule 'space-after-keywords' was removed and replaced by: keyword-spacing","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"space-return-throw-case","severity":1,"message":"Rule 'space-return-throw-case' was removed and replaced by: keyword-spacing","line":1,"column":1,"nodeType":null,"source":"import React, { Component } from 'react';"},{"ruleId":"no-unused-vars","severity":1,"message":"'Component' is defined but never used.","line":1,"column":17,"nodeType":"Identifier","source":"import React, { Component } from 'react';","endLine":1,"endColumn":26}],"errorCount":0,"warningCount":7,"fixableErrorCount":0,"fixableWarningCount":0,"source":"import React, { Component } from 'react';\nimport './App.css';\n\nconst App = () => {\n  return (\n    <div className=\"App\">Hello, world!</div>\n  );\n};\n\nexport default App;\n"}]
SublimeLinter: #2 SublimeLinter-eslint.sublime-package:56: eslint output:
[{'errorCount': 0,
  'filePath': '/home/daniel/dev/react/hello-world/src/App.js',
  'fixableErrorCount': 0,
  'fixableWarningCount': 0,
  'messages': [{'column': 1,
                'line': 1,
                'message': "Definition for rule 'react/jsx-quotes' was not found",
                'nodeType': None,
                'ruleId': 'react/jsx-quotes',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 1,
                'line': 1,
                'message': "Definition for rule 'react/require-extension' was not found",
                'nodeType': None,
                'ruleId': 'react/require-extension',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 1,
                'line': 1,
                'message': "Definition for rule 'react/wrap-multilines' was not found",
                'nodeType': None,
                'ruleId': 'react/wrap-multilines',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 1,
                'line': 1,
                'message': "Rule 'no-empty-label' was removed and replaced by: no-labels",
                'nodeType': None,
                'ruleId': 'no-empty-label',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 1,
                'line': 1,
                'message': "Rule 'space-after-keywords' was removed and replaced by: keyword-spacing",
                'nodeType': None,
                'ruleId': 'space-after-keywords',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 1,
                'line': 1,
                'message': "Rule 'space-return-throw-case' was removed and replaced by: keyword-spacing",
                'nodeType': None,
                'ruleId': 'space-return-throw-case',
                'severity': 1,
                'source': "import React, { Component } from 'react';"},
               {'column': 17,
                'endColumn': 26,
                'endLine': 1,
                'line': 1,
                'message': "'Component' is defined but never used.",
                'nodeType': 'Identifier',
                'ruleId': 'no-unused-vars',
                'severity': 1,
                'source': "import React, { Component } from 'react';"}],
  'source': 'import React, { Component } from \'react\';\nimport \'./App.css\';\n\nconst App = () => {\n  return (\n    <div className="App">Hello, world!</div>\n  );\n};\n\nexport default App;\n',
  'warningCount': 7}]
Package Control: Skipping automatic upgrade, last run at 2018-03-07 14:37:49, next run at 2018-03-07 15:37:49 or after
worker 5448 appears stuck while processing file /home/daniel/dev/react/hello-world/node_modules/diff/dist/diff.js, killing process
indexing: crawler exited while processing /home/daniel/dev/react/hello-world/node_modules/diff/dist/diff.js, no symbols recorded
```

**Comment by braver** [comment]
> I believe this is a problem from your linter since executing the eslint command doesn't give this error

> SublimeLinter: #1 SublimeLinter.sublime-package:809: jshint output:
    /home/daniel/dev/react/hello-world/src/App.js: line 6, col 40, Unclosed regular expression. (E015)
    /home/daniel/dev/react/hello-world/src/App.js: line 6, col 40, Unrecoverable syntax error. (54% scanned). (E041)
>
>    2 errors

One is `eslint`, the other is `jshint`?

**Comment by alexandre1985** [comment]
Here is a screenshot.
![deepinscreenshot_select-area_20180307144510](https://user-images.githubusercontent.com/5586308/37098642-4cb00e00-2216-11e8-8430-7edafed6025b.png)

**Comment by alexandre1985** [comment]
I have jshint disabled.
Here is my SublimeLinter User Settings:
```
// SublimeLinter Settings - User
{
	"linters":{
		"eslint":{
			"excludes":[
				"dist/*",
				"node_modules/*"
			]
		},
		"jshint":{
			"@disable":true
		}
	},
	"paths": {
		"linux": ["~/n/bin/"],
        "osx": [],
        "windows": []
	},
	"debug": true
}

```

**Comment by braver** [comment]
Ah, well that's it then. From the [upgrade message](https://github.com/SublimeLinter/SublimeLinter/blob/master/messages/4.0.0.txt):

> Note that "@disable" is now "disable", without the "@".

**Comment by alexandre1985** [comment]
Thank you. It solved my issue. =) I'm happy

**Comment by braver** [comment]
w00t! 😀
